src/Functions/HasTokenImpl.h

```diff
@@ -46,10 +47,24 @@ struct HasTokenImpl
         /// The current index in the array of strings.
         size_t i = 0;
 
-        TokenSearcher searcher(pattern.data(), pattern.size(), end - pos);
+        std::optional<TokenSearcher> searcher;
```

Comment:
It's not clear why TokenSearcher is now wrapped in std::optional.

Reply:
Because whether or not it's going to have a value depends on the outcome of the try-catch block. Since in the cases it'd result in `std::nullopt` are scenarios when either an exception is thrown or null returned, I can take a different approach and eliminate this.